src/realm/sync/changeset_encoder.cpp

```diff
@@ -206,9 +206,10 @@ void ChangesetEncoder::operator()(const Instruction::ArrayErase& instr)
     append_path_instr(Instruction::Type::ArrayErase, instr, instr.prior_size);
 }
 
-void ChangesetEncoder::operator()(const Instruction::ArrayClear& instr)
+void ChangesetEncoder::operator()(const Instruction::Clear& instr)
 {
-    append_path_instr(Instruction::Type::ArrayClear, instr, instr.prior_size);
+    uint32_t prior_size = 0; // Ignored
```

Comment:
Why do this? Are we officially just not caring about this field at all?

Reply:
Yeah. The issue that this fixes is that we failed to maintain the value in the rule ArrayClear/ArrayInsert. In general, I no longer believe that it is a useful thing to keep around.

Reply:
The alternative would be to insert some if statements in the nested merge rule for ArrayClear to detect if it meets an ArrayErase or ArrayInsert instruction, and inc/dec the prior_size accordingly.